app/TuistApp/Sources/Views/DevicesView/DevicesView.swift

```diff
@@ -94,11 +64,98 @@ struct DevicesView: View, ErrorViewHandling {
                     }
                 }
             }
-            .padding([.leading, .trailing], 8)
         }
         .frame(height: isExpanded ? NSScreen.main.map { $0.visibleFrame.size.height - 300 } ?? 500 : nil)
     }
 
+    @ViewBuilder
+    private func pinnedDevicesSection() -> some View {
+        if !viewModel.devices.isEmpty || !viewModel.pinnedSimulators.isEmpty {
+            HStack {
+                Text("Devices")
+                    .font(.headline)
+                    .fontWeight(.medium)
+                    .padding(.leading, 4)
+
+                Spacer()
+
+                Button {
+                    withAnimation(.linear(duration: 0.8)) {
+                        isRefreshing = true
+                        Task {
+                            try await viewModel.onAppear()
+                            await MainActor.run {
+                                isRefreshing = false
+                            }
```

Comment:
If `viewModel.onAppear` throws, this is never called afaict. I'd consider moving this to a view model as well and test these scenarios.